packages/gatsby/src/bootstrap/load-plugins/load.js

```diff
@@ -243,6 +243,21 @@ const loadPlugins = (config = {}, rootDir = null) => {
     }
   }
 
+  // TypeScript support by default! use the user-provided one if it exists
+  const typescriptPlugin = (config.plugins || []).find(
+    plugin =>
+      plugin.resolve === `gatsby-plugin-typescript` ||
+      plugin === `gatsby-plugin-typescript`
```

Comment:
Can we ever end up in scenario where `plugin` is actually a `string` and not ` { resolve: string, options: record<string, unknown> }`?

I guess until we have this and rest of code converted to typescript this is just safe thing to do and I'm fine with it

Reply:
Ah -- yeah, aren't these always merged into this shape even if we're starting with a string? Good point.

Reply:
I guess those can be strings actually so this is correct if we are checking `config.plugins`

Those would be guaranteed objects if expected shape if we would be checking local `plugins` array (as we push "processed plugin" entries there)

Reply:
I'm not sure. But I sort of assumed that these 2 could be triggered if I create a config like

```
exports = {
  plugins: [
    `gatsby-plugin-typescript`,
    { resolve: `gatsby-plugin-typescript` }
  ]
}
```